core/domain/activity_jobs_one_off.py

```diff
@@ -589,3 +589,103 @@ def map(snapshot_model):
     def reduce(key, values):
         """Implements the reduce function for this job."""
         yield (key, len(values))
+
+
+class ValidateSnapshotMetadataModelsJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that validates the commit logs of the SnapshotMetadataModels.
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        # Considering all the models that has a SnapshotMetadata except
+        # config model because it does not have commit logs.
+        return [collection_models.CollectionSnapshotMetadataModel,
+                exp_models.ExplorationRightsSnapshotMetadataModel,
+                question_models.QuestionSnapshotMetadataModel,
+                skill_models.SkillSnapshotMetadataModel,
+                story_models.StorySnapshotMetadataModel,
+                topic_models.TopicSnapshotMetadataModel,
+                topic_models.SubtopicPageSnapshotMetadataModel,
+                topic_models.TopicRightsSnapshotMetadataModel]
+
+    @staticmethod
+    def map(snapshot_model):
+        if snapshot_model.deleted:
+            return
```

Comment:
Please at least log these cases.

Reply:
I have removed this case. Now I am checking the existence of the parent class.